<a href="https://colab.research.google.com/github/alicefing/Alice-Fin---Exerc-cios-de-programa-o/blob/main/Extra_2_Pensamento_Computacional.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>


<html>
  <body>
    <header></header>
        <CENTER>
          <img src="https://uvv.br/wp-content/themes/core/dist/images/Logo.png" alt="UVV-LOGO" style = width="100px"; height="100px">
        </CENTER>
        <CENTER><b>Fundamentos de Tecnologia da Computação</b></CENTER>
        <CENTER><b>Atividade Extra</b></CENTER>

NOME:

MEU LINK COLAB: [Link]()

Dado o algoritmo em Python da situação-problema: equação do 2º grau, a seguir:

In [ ]:
try:
  A = float(input('Coeficiente A: '))
  B = float(input('Coeficiente B: '))
  C = float(input('Coeficiente C: '))
  if A == 0 or (B == 0 and C == 0):
    print('Não é uma Equação do 2° Grau.')
  else:
    delta = B ** 2 - 4 * A * C
    print(f'Valor de Delta: {delta: .2f}')
    if delta < 0:
      print('Não tem Raízes no conjunto dos números reais.')
    elif delta == 0:
      X1 = -B / (2 * A)
      print(f'Raiz X1 = X2 = {X1: .2f}')
    else:
      raizDelta = delta ** 0.5
      print(f'Valor da Raíz de Delta: {raizDelta: .2f}')
      X1 = (-B + raizDelta) / (2 * A)
      X2 = (-B - raizDelta) / (2 * A)
      print(f'Raiz X1 = {X1: .2f}')
      print(f'Raiz X2 = {X2: .2f}')
except Exception as ERRO_EXCECAO:
  print(f'Erro de exceção: {ERRO_EXCECAO}')

Coeficiente A: 0
Coeficiente B: 1
Coeficiente C: 2
Não é uma Equação do 2° Grau.


Com isso, copiar e execute o algoritmo acima **5 vezes**, escolhendo os coeficientes $A$, $B$ e $C$ para as 5 grupos de soluções por decomposição:

# DECOMPOSIÇÃO 1:

Os coeficientes A, B ou C não formam uma equação do 2º grau:

In [1]:
# Executar seu algoritmo aqui:
try:
  A = float(input('Coeficiente A: '))
  B = float(input('Coeficiente B: '))
  C = float(input('Coeficiente C: '))
  if A == 0 or (B == 0 and C == 0):
    print('Não é uma Equação do 2° Grau.')
  else:
    delta = B ** 2 - 4 * A * C
    print(f'Valor de Delta: {delta: .2f}')
    if delta < 0:
      print('Não tem Raízes no conjunto dos números reais.')
    elif delta == 0:
      X1 = -B / (2 * A)
      print(f'Raiz X1 = X2 = {X1: .2f}')
    else:
      raizDelta = delta ** 0.5
      print(f'Valor da Raíz de Delta: {raizDelta: .2f}')
      X1 = (-B + raizDelta) / (2 * A)
      X2 = (-B - raizDelta) / (2 * A)
      print(f'Raiz X1 = {X1: .2f}')
      print(f'Raiz X2 = {X2: .2f}')
except Exception as ERRO_EXCECAO:
  print(f'Erro de exceção: {ERRO_EXCECAO}')

Coeficiente A: 0
Coeficiente B: 0
Coeficiente C: 0
Não é uma Equação do 2° Grau.


Programar na próxima célula, usando as linguagens de marcação HTML + CSS + LATEX, a equação montada com os valores:

*   Coeficientes $A$, $B$ e $C$.
*   Delta e Raiz de Delta, se existir.
* $x_1$ e $x_2$, se existir.

<!DOCTYPE html>
<html lang="pt-br">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>DECOMPOSIÇÃO 1 ----------------------------------------------------------------------------------------------------------------------------</title>
</head>
<body>
    <h3>Resolução da Equação do Segundo Grau</h3>
<br>
    - Coeficientes testados ---->  A=0; B=0; C=0
$$ 0x^2 + 0x + 0 = 0 $$
<br>
     - DELTA:  O valor de delta é nulo, visto que todos os coeficientes tem o valor 0. No código, nem calcula delta pois antes o programa já entende que não será possível pelos valores dos coeficientes.
$$ \Delta = 0^2 - 4*0*0 $$
<br>
    - RAÍZES: Não é possível ter raizes reais nesse caso em que delta é menor que zero.

</body>
</html>





RASTREAMENTO DO ALGORITMO: $$ 0x^2 + 0x + 0 = 0 $$

*   1º PASSO (AÇÕES ou DECISÕES = ação = LER: Coeficiente $a = 0$)
*   2º PASSO (AÇÕES ou DECISÕES = ação = LER: Coeficiente $b = 0$)
*   3º PASSO (AÇÕES ou DECISÕES = ação = LER: Coeficiente $c = 0$)
*   4º PASSO (AÇÕES ou DECISÕES = decisão = EXIBIR: 'Não é uma Equação do 2° Grau.')


# DECOMPOSIÇÃO 2:

Os coeficientes A, B ou C formam uma equação do 2º grau: de delta < 0:

In [2]:
# Executar seu algoritmo aqui:
try:
  A = float(input('Coeficiente A: '))
  B = float(input('Coeficiente B: '))
  C = float(input('Coeficiente C: '))
  if A == 0 or (B == 0 and C == 0):
    print('Não é uma Equação do 2° Grau.')
  else:
    delta = B ** 2 - 4 * A * C
    print(f'Valor de Delta: {delta: .2f}')
    if delta < 0:
      print('Não tem Raízes no conjunto dos números reais.')
    elif delta == 0:
      X1 = -B / (2 * A)
      print(f'Raiz X1 = X2 = {X1: .2f}')
    else:
      raizDelta = delta ** 0.5
      print(f'Valor da Raíz de Delta: {raizDelta: .2f}')
      X1 = (-B + raizDelta) / (2 * A)
      X2 = (-B - raizDelta) / (2 * A)
      print(f'Raiz X1 = {X1: .2f}')
      print(f'Raiz X2 = {X2: .2f}')
except Exception as ERRO_EXCECAO:
  print(f'Erro de exceção: {ERRO_EXCECAO}')


Coeficiente A: 2
Coeficiente B: -1
Coeficiente C: 3
Valor de Delta: -23.00
Não tem Raízes no conjunto dos números reais.


Programar na próxima célula, usando as linguagens de marcação HTML + CSS + LATEX, a equação montada com os valores:

*   Coeficientes $A$, $B$ e $C$.
*   Delta e Raiz de Delta, se existir.
* $x_1$ e $x_2$, se existir.

<!DOCTYPE html>
<html lang="pt-br">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>DECOMPOSIÇÃO 2 ----------------------------------------------------------------------------------------------------------------------------</title>
</head>
<body>
    <h3>Resolução da Equação do Segundo Grau</h3>
<br>
    - Coeficientes testados ---->  A=2; B=-1; C=3
$$ 2x^2 -x + 3 = 0 $$
<br>
     - DELTA:  O valor de delta é negativo, visto que o coenficiente que multiplica que B e é elevado a 2 é negativo. A e C, que são multiplicados ao -4, são numero poisitivos. No código, quando multiplicado todos os valores, DELTA é menor que zero.
$$ \Delta = -1^2 - 4*2*3 $$
$$ \Delta = -23 $$
<br>
    - RAÍZES: Não tem Raízes no conjunto dos números reais.

</body>
</html>


RASTREAMENTO DO ALGORITMO: $$ 2x^2 - x + 3 = 0 $$

*   1º PASSO (AÇÕES ou DECISÕES = ação = LER: Coeficiente $a = 2$)
*   2º PASSO (AÇÕES ou DECISÕES = ação = LER: Coeficiente $b = -1$)
*   3º PASSO (AÇÕES ou DECISÕES = ação = LER: Coeficiente $c = 3$)
*   4º PASSO (AÇÕES ou DECISÕES = DECISÃO = EXIBIR: 'Valor de Delta: -23.00')
*   5º PASSO (AÇÕES ou DECISÕES = DECISÃO = EXIBIR: 'Não tem Raízes no conjunto dos números reais.')



# DECOMPOSIÇÃO 3:

Os coeficientes A, B ou C formam uma equação do 2º grau: de delta = 0:

In [3]:
# Executar seu algoritmo aqui:
try:
  A = float(input('Coeficiente A: '))
  B = float(input('Coeficiente B: '))
  C = float(input('Coeficiente C: '))
  if A == 0 or (B == 0 and C == 0):
    print('Não é uma Equação do 2° Grau.')
  else:
    delta = B ** 2 - 4 * A * C
    print(f'Valor de Delta: {delta: .2f}')
    if delta < 0:
      print('Não tem Raízes no conjunto dos números reais.')
    elif delta == 0:
      X1 = -B / (2 * A)
      print(f'Raiz X1 = X2 = {X1: .2f}')
    else:
      raizDelta = delta ** 0.5
      print(f'Valor da Raíz de Delta: {raizDelta: .2f}')
      X1 = (-B + raizDelta) / (2 * A)
      X2 = (-B - raizDelta) / (2 * A)
      print(f'Raiz X1 = {X1: .2f}')
      print(f'Raiz X2 = {X2: .2f}')
except Exception as ERRO_EXCECAO:
  print(f'Erro de exceção: {ERRO_EXCECAO}')


Coeficiente A: 1
Coeficiente B: 2
Coeficiente C: 1
Valor de Delta:  0.00
Raiz X1 = X2 = -1.00


Programar na próxima célula, usando as linguagens de marcação HTML + CSS + LATEX, a equação montada com os valores:

*   Coeficientes $A$, $B$ e $C$.
*   Delta e Raiz de Delta, se existir.
* $x_1$ e $x_2$, se existir.

<!DOCTYPE html>
<html lang="pt-br">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>DECOMPOSIÇÃO 3 ----------------------------------------------------------------------------------------------------------------------------</title>
</head>
<body>
    <h3>Resolução da Equação do Segundo Grau</h3>
<br>
    - Coeficientes testados ---->  A=1; B=2; C=1
$$ x^2 + 2 + 1 = 0 $$
<br>
     - DELTA:  O valor de delta é igual a zero, visto que o coenficiente que multiplica que B resulta em 4. A multiplica -4 que resulta -4 e C multiplica 0 -4, que resulta em -4 No código, quando multiplicado todos os valores, DELTA é igual a zero.
$$ \Delta = 2^2 - 4*1*1 $$
$$ \Delta = 0 $$
<br>
    - RAÍZES: Raiz X1 = X2 = -1.00

</body>
</html>

RASTREAMENTO DO ALGORITMO: $$ x^2 + 2 + 1 = 0 $$

*   1º PASSO (AÇÕES ou DECISÕES = ação = LER: Coeficiente $a = 1$)
*   2º PASSO (AÇÕES ou DECISÕES = ação = LER: Coeficiente $b = 2$)
*   3º PASSO (AÇÕES ou DECISÕES = açãp = LER: Coeficiente $c = 1$)
*   4º PASSO (AÇÕES ou DECISÕES = decisão = EXIBIR: 'Valor de Delta:  0.00')
*   5º PASSO (AÇÕES ou DECISÕES = decisão = EXIBIR: 'Valor de Delta:  Raiz X1 = X2 = -1.00
')


# DECOMPOSIÇÃO 4:

Os coeficientes A, B ou C formam uma equação do 2º grau: de delta > 0:

In [6]:
# Executar seu algoritmo aqui:
try:
  A = float(input('Coeficiente A: '))
  B = float(input('Coeficiente B: '))
  C = float(input('Coeficiente C: '))
  if A == 0 or (B == 0 and C == 0):
    print('Não é uma Equação do 2° Grau.')
  else:
    delta = B ** 2 - 4 * A * C
    print(f'Valor de Delta: {delta: .2f}')
    if delta < 0:
      print('Não tem Raízes no conjunto dos números reais.')
    elif delta == 0:
      X1 = -B / (2 * A)
      print(f'Raiz X1 = X2 = {X1: .2f}')
    else:
      raizDelta = delta ** 0.5
      print(f'Valor da Raíz de Delta: {raizDelta: .2f}')
      X1 = (-B + raizDelta) / (2 * A)
      X2 = (-B - raizDelta) / (2 * A)
      print(f'Raiz X1 = {X1: .2f}')
      print(f'Raiz X2 = {X2: .2f}')
except Exception as ERRO_EXCECAO:
  print(f'Erro de exceção: {ERRO_EXCECAO}')


Coeficiente A: 2
Coeficiente B: 4
Coeficiente C: 1
Valor de Delta:  8.00
Valor da Raíz de Delta:  2.83
Raiz X1 = -0.29
Raiz X2 = -1.71


Programar na próxima célula, usando as linguagens de marcação HTML + CSS + LATEX, a equação montada com os valores:

*   Coeficientes $A$, $B$ e $C$.
*   Delta e Raiz de Delta, se existir.
* $x_1$ e $x_2$, se existir.

<!DOCTYPE html>
<html lang="pt-br">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>DECOMPOSIÇÃO 4 ----------------------------------------------------------------------------------------------------------------------------</title>
</head>
<body>
    <h3>Resolução da Equação do Segundo Grau</h3>
<br>
    - Coeficientes testados ---->  A=2; B=4; C=1
$$ 2x^2 + 4 + 1 = 0 $$
<br>
     - DELTA:  O valor de delta é maior que zero, visto que o coenficiente que multiplica que B resulta em 16. A multiplica -4 que resulta -8 e C valendo 1 multiplica o -8 , que resulta em -8. No código, quando multiplicado todos os valores, DELTA é maior que zero.
$$ \Delta = 4^2 - 4*2*1 $$
$$ \Delta = 8 $$
<br>
    - RAÍZES: Raiz X1 = -0.29 | Raiz X2 = -1.71

</body>
</html>

RRASTREAMENTO DO ALGORITMO: $$ 2x^2 + 4 + 1 = 0 $$

*   1º PASSO (AÇÕES ou DECISÕES = ação = LER: Coeficiente $a = 2$)
*   2º PASSO (AÇÕES ou DECISÕES = ação = LER: Coeficiente $b = 4$)
*   3º PASSO (AÇÕES ou DECISÕES = ação = LER: Coeficiente $c = 1$)
*   4º PASSO (AÇÕES ou DECISÕES = decisão = EXIBIR: 'Valor de Delta:  8.00')
*   5º PASSO (AÇÕES ou DECISÕES = decisão = EXIBIR: 'Valor da Raíz de Delta:  2.83')
*   6º PASSO (AÇÕES ou DECISÕES = decisão = EXIBIR: Raiz X1 = -0.29
Raiz X2 = -1.71
')


# DECOMPOSIÇÃO 5:

Os coeficientes A, B ou C levam o algoritmo a um **Erro de Exceção** (vai executar o comando **except**):

In [ ]:
try:
  A = float(input('Coeficiente A: '))
  B = float(input('Coeficiente B: '))
  C = float(input('Coeficiente C: '))
  if A == 0 or (B == 0 and C == 0):
    print('Não é uma Equação do 2° Grau.')
  else:
    delta = B ** 2 - 4 * A * C
    print(f'Valor de Delta: {delta: .2f}')
    if delta < 0:
      print('Não tem Raízes no conjunto dos números reais.')
    elif delta == 0:
      X1 = -B / (2 * A)
      print(f'Raiz X1 = X2 = {X1: .2f}')
    else:
      raizDelta = delta ** 0.5
      print(f'Valor da Raíz de Delta: {raizDelta: .2f}')
      X1 = (-B + raizDelta) / (2 * A)
      X2 = (-B - raizDelta) / (2 * A)
      print(f'Raiz X1 = {X1: .2f}')
      print(f'Raiz X2 = {X2: .2f}')
except Exception as ERRO_EXCECAO:
  print(f'Erro de exceção: {ERRO_EXCECAO}')

Coeficiente A: 3.14
Coeficiente B: -5
Coeficiente C: -1,3
Erro de exceção: could not convert string to float: '-1,3'


Programar na próxima célula, usando as linguagens de marcação HTML + CSS + LATEX, faça o RASTREAMENTO (PASSO A PASSO: AÇÕES + DECISÕES) a equação montada com os valores:

*   Coeficientes $A$, $B$ e / ou $C$.
*   Delta e Raiz de Delta, se existir.
* $x_1$ e $x_2$, se existir.

<!DOCTYPE html>
<html lang="pt-br">
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>DECOMPOSIÇÃO 5 ----------------------------------------------------------------------------------------------------------------------------</title>
</head>
<body>
    <h3>Resolução da Equação do Segundo Grau</h3>
<br>
    - Coeficientes testados ---->  A=3.14; B=-5; C=-1,3
$$\pi x^2 - 5x -1,3 = 0$$
<br>
     - DELTA:  Não se calcula o delta porque não é possível converter a string para float.


<br>
    - RAÍZES: Não se calcula a raiz pois não é possível ter um delta, visto que não é possível convertar a string para float.

</body>
</html>

RASTREAMENTO DO ALGORITMO: $\pi$ x^2 - 5x -1,3 = 0$

*   1º PASSO (AÇÕES ou DECISÕES = ação = LER: Coeficiente $a = \pi$)
*   2º PASSO (AÇÕES ou DECISÕES = ação = LER: Coeficiente $b = -5$)
*   3º PASSO (AÇÕES ou DECISÕES = ação = LER: Coeficiente $c = -1,3$)
*   4º PASSO (AÇÕES ou DECISÕES = ação = EXIBIR: 'Erro de exceção: could not convert string to float: '-1,3'')